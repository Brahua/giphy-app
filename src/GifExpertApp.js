import React, { useState } from 'react'
import GifGrid from './components/GifGrid.jsx';
import AddCategory from './components/AddCategory.jsx';

export default function GifExpertApp({defaultCategories = ['DBZ']}) {

    const [categories, setCategories] = useState(defaultCategories);
  
    return (
      <>
        <div className="jumbotron jumbotron-fluid bg-color-jumbotron">
          <div className="container">
            <h1 className="display-4">GiphyApp</h1>
            <p className="lead">This is a React app</p>
          </div>
        </div>
        <div className="container">
          <AddCategory setCategories={setCategories}/>
          <hr/>   
          { categories.map(category => <GifGrid key={category} category={category}/>) }
        </div> 
      </>
    )
  
}
