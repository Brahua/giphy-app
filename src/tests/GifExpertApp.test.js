import React from 'react';
import { shallow } from 'enzyme';
import GifExpertApp from './../GifExpertApp';
import '@testing-library/jest-dom'

describe('Tests in <GifExpertApp>', () => {
  test('should render correctly', () => {
    const wrapper = shallow(<GifExpertApp/>);
    expect(wrapper).toMatchSnapshot();
  });

  test('should show an array of gifÂ´s categories', () => {
    const categories = ['Naruto', 'Dragon Ball'];
    const wrapper = shallow(<GifExpertApp defaultCategories={categories}/>);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('GifGrid').length).toBe(categories.length);
  });
})
