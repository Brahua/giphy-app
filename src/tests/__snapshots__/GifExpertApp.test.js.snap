// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <GifExpertApp> should render correctly 1`] = `
<Fragment>
  <div
    className="jumbotron jumbotron-fluid bg-color-jumbotron"
  >
    <div
      className="container"
    >
      <h1
        className="display-4"
      >
        GiphyApp
      </h1>
      <p
        className="lead"
      >
        This is a React app
      </p>
    </div>
  </div>
  <div
    className="container"
  >
    <AddCategory
      setCategories={[Function]}
    />
    <hr />
    <GifGrid
      category="DBZ"
      key="DBZ"
    />
  </div>
</Fragment>
`;

exports[`Tests in <GifExpertApp> should show an array of gifÂ´s categories 1`] = `
<Fragment>
  <div
    className="jumbotron jumbotron-fluid bg-color-jumbotron"
  >
    <div
      className="container"
    >
      <h1
        className="display-4"
      >
        GiphyApp
      </h1>
      <p
        className="lead"
      >
        This is a React app
      </p>
    </div>
  </div>
  <div
    className="container"
  >
    <AddCategory
      setCategories={[Function]}
    />
    <hr />
    <GifGrid
      category="Naruto"
      key="Naruto"
    />
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
  </div>
</Fragment>
`;
